source-code/core2/src/messages/query.ts

```diff
@@ -1,15 +1,41 @@
 import type { Message } from "./schema.js"
 
-// TODO
-// 1. Query builder?
-// 2. Filtering etc how?
-//
-// inspiration: Prisma CRUD https://www.prisma.io/docs/concepts/components/prisma-client/crud
-export type MessagesQueryApi_1 = {
-	create: (message: Message) => void
-	get: () => Message[]
-	update: () => void
-	delete: () => void
+export type MessageQueryApi = {
+	create: (args: { data: Message }) => void
+	get: (args: { where: { id: Message["id"] } }) => Message | undefined
```

Comment:
How can we access all messages? Are we in the future adding filters to the get query to return, for example, only one variant or only one language? To achieve something similar from outside, we need at least access to all the messages.

Reply:
We gonna discuss an `includedMessageIds`function in a follow-up PR for accessing all the messages.